main/lsp/requests/hover.cc

```diff
@@ -89,9 +89,15 @@ LSPResult LSPLoop::handleTextDocumentHover(unique_ptr<core::GlobalState> gs, con
             if (constraint) {
                 retType = core::Types::instantiate(core::Context(*gs, core::Symbols::root()), retType, *constraint);
             }
-            auto methodInfo = methodInfoString(*gs, retType, *sendResp->dispatchResult, constraint);
-            response->result =
-                make_unique<Hover>(formatHoverText(config.clientHoverMarkupKind, methodInfo, documentation));
+            if (sendResp->dispatchResult->main.method.exists() && sendResp->dispatchResult->main.method.isSynthetic()) {
```

Comment:
Interesting. I looked into what it means for a Symbol to be marked "synthetic"; it looks like it just means that it's entered in GlobalState... that might end up biting us in the future if we entered a method symbol in GlobalState that was not synthetic. In particular, it's currently the case that every *intrinsic* method symbol is already installed **after** the "isSynthetic" region of symbols, which is what I originally had in mind that would break this.

So yeah, no change expected, just wanted to share the result of my digging into the meaning of `isSynthetic` in this context.

Reply:
@jez what do you mean? I thought synthetic methods are those fake methods ginned up by Sorbet internally. There are plenty of symbols entered in GlobalState that aren't synthetic.